In [2]:
import numpy as np
import pandas as pd
from scipy import stats

# Hypothesis Testing

## Week 4 Day 4

This notebook demonstrates hypothesis testing using three business scenarios.

## Scenario 1: One-Sample T-Test

### Business Question

A company has a target average delivery time of 5 days. We want to determine whether the actual average delivery time is significantly different from 5 days.

### Hypotheses

- **Null Hypothesis (H₀):** The average delivery time is equal to 5 days.
- **Alternative Hypothesis (H₁):** The average delivery time is different from 5 days.

**Significance Level (α): 0.05**

In [4]:
delivery_days = np.array([5.2, 4.8, 5.5, 5.1, 4.9, 5.3, 5.0, 4.7, 5.4, 5.2])

t_stat, p_value = stats.ttest_1samp(delivery_days, popmean=5)

confidence_interval = stats.t.interval(
    0.95,
    df=len(delivery_days) - 1,
    loc=np.mean(delivery_days),
    scale=stats.sem(delivery_days)
)

print("Sample Mean:", np.mean(delivery_days))
print("T-statistic:", t_stat)
print("P-value:", p_value)
print("95% Confidence Interval:", confidence_interval)

Sample Mean: 5.11
T-statistic: 1.3372274824806316
P-value: 0.2139601131740449
95% Confidence Interval: (np.float64(4.923915514624187), np.float64(5.296084485375814))


In [5]:
if p_value < 0.05:
    print("Decision: Reject the null hypothesis.")
    print("Interpretation: The average delivery time is significantly different from 5 days.")
else:
    print("Decision: Fail to reject the null hypothesis.")
    print("Interpretation: There is not enough evidence to conclude that the average delivery time is different from 5 days.")

Decision: Fail to reject the null hypothesis.
Interpretation: There is not enough evidence to conclude that the average delivery time is different from 5 days.


### Interpretation

The p-value is 0.2140, which is greater than the significance level of 0.05. Therefore, we fail to reject the null hypothesis.

There is not enough statistical evidence to conclude that the average delivery time is different from the target of 5 days.

The 95% confidence interval for the average delivery time is approximately 4.924 to 5.296 days.

## Scenario 2: Two-Sample T-Test

### Business Question

A company wants to compare the average sales of two regions to determine whether their average sales are significantly different.

### Hypotheses

- **Null Hypothesis (H₀):** The average sales of Region A and Region B are equal.
- **Alternative Hypothesis (H₁):** The average sales of Region A and Region B are different.

**Significance Level (α): 0.05**

In [6]:
region_a_sales = np.array([520, 540, 510, 530, 550, 525, 515, 545, 535, 528])
region_b_sales = np.array([480, 495, 470, 500, 485, 490, 475, 505, 492, 488])

t_stat_2, p_value_2 = stats.ttest_ind(
    region_a_sales,
    region_b_sales,
    equal_var=True
)

print("Region A Mean:", np.mean(region_a_sales))
print("Region B Mean:", np.mean(region_b_sales))
print("T-statistic:", t_stat_2)
print("P-value:", p_value_2)

Region A Mean: 529.8
Region B Mean: 488.0
T-statistic: 7.819803282592298
P-value: 3.3848274864012303e-07


In [7]:
if p_value_2 < 0.05:
    print("Decision: Reject the null hypothesis.")
    print("Interpretation: There is a statistically significant difference between the average sales of Region A and Region B.")
else:
    print("Decision: Fail to reject the null hypothesis.")
    print("Interpretation: There is not enough evidence to conclude that the average sales of Region A and Region B are different.")

Decision: Reject the null hypothesis.
Interpretation: There is a statistically significant difference between the average sales of Region A and Region B.


### Interpretation

The p-value is approximately 0.000000338, which is less than the significance level of 0.05. Therefore, we reject the null hypothesis.

There is statistically significant evidence that the average sales of Region A and Region B are different.

Region A has an average sales value of 529.8, while Region B has an average sales value of 488.0.

## Scenario 3: Customer Spending

### Business Question

A company historically expects customers to spend an average of ₹1,000 per transaction. We want to determine whether the current average customer spending is significantly different from ₹1,000.

### Hypotheses

- **Null Hypothesis (H₀):** The average customer spending is equal to ₹1,000.
- **Alternative Hypothesis (H₁):** The average customer spending is different from ₹1,000.

**Significance Level (α): 0.05**

In [8]:
customer_spending = np.array([
    950, 1020, 980, 1100, 1050,
    990, 1010, 970, 1080, 1040
])

t_stat_3, p_value_3 = stats.ttest_1samp(
    customer_spending,
    popmean=1000
)

confidence_interval_3 = stats.t.interval(
    0.95,
    df=len(customer_spending) - 1,
    loc=np.mean(customer_spending),
    scale=stats.sem(customer_spending)
)

print("Sample Mean:", np.mean(customer_spending))
print("T-statistic:", t_stat_3)
print("P-value:", p_value_3)
print("95% Confidence Interval:", confidence_interval_3)

Sample Mean: 1019.0
T-statistic: 1.2353414818569877
P-value: 0.2479781240442151
95% Confidence Interval: (np.float64(984.207202442068), np.float64(1053.792797557932))


## Scenario 3: Customer Spending

### Business Question

A company historically expects customers to spend an average of ₹1,000 per transaction. We want to determine whether the current average customer spending is significantly different from ₹1,000.

### Hypotheses

- **Null Hypothesis (H₀):** The average customer spending is equal to ₹1,000.
- **Alternative Hypothesis (H₁):** The average customer spending is different from ₹1,000.

**Significance Level (α): 0.05**

In [9]:
customer_spending = np.array([
    950, 1020, 980, 1100, 1050,
    990, 1010, 970, 1080, 1040
])

t_stat_3, p_value_3 = stats.ttest_1samp(
    customer_spending,
    popmean=1000
)

confidence_interval_3 = stats.t.interval(
    0.95,
    df=len(customer_spending) - 1,
    loc=np.mean(customer_spending),
    scale=stats.sem(customer_spending)
)

print("Sample Mean:", np.mean(customer_spending))
print("T-statistic:", t_stat_3)
print("P-value:", p_value_3)
print("95% Confidence Interval:", confidence_interval_3)

Sample Mean: 1019.0
T-statistic: 1.2353414818569877
P-value: 0.2479781240442151
95% Confidence Interval: (np.float64(984.207202442068), np.float64(1053.792797557932))


In [10]:
if p_value_3 < 0.05:
    print("Decision: Reject the null hypothesis.")
    print("Interpretation: The average customer spending is significantly different from ₹1,000.")
else:
    print("Decision: Fail to reject the null hypothesis.")
    print("Interpretation: There is not enough evidence to conclude that the average customer spending is different from ₹1,000.")

Decision: Fail to reject the null hypothesis.
Interpretation: There is not enough evidence to conclude that the average customer spending is different from ₹1,000.


### Interpretation

The p-value is compared with the significance level of 0.05. If the p-value is less than 0.05, we reject the null hypothesis. Otherwise, we fail to reject the null hypothesis.

The 95% confidence interval provides a range of plausible values for the population mean customer spending.

## Type I and Type II Errors

### Type I Error

A Type I error occurs when we reject the null hypothesis even though it is actually true.

**Business consequence:** A company may conclude that a strategy, product, or process has a significant effect when there is actually no real effect. This could lead to unnecessary spending or business changes.

### Type II Error

A Type II error occurs when we fail to reject the null hypothesis even though it is actually false.

**Business consequence:** A company may fail to identify a real improvement or important difference, potentially causing it to miss a useful business opportunity.

## Conclusion

Hypothesis testing provides a statistical framework for evaluating business assumptions. One-sample and two-sample t-tests can be used to compare sample means with a benchmark or between two independent groups. P-values and confidence intervals help interpret the statistical evidence, while understanding Type I and Type II errors helps identify potential business consequences of incorrect decisions.